### PR TITLE
Split `install_R.sh` into two scripts and some improvements in the R installation process

### LINKED ISSUE #385: Set CRAN url after installing R
Currently, the Dockerfile to install R has the following structure.
The `CRAN` environment variable for setting the CRAN mirror is set before R is installed, so the cache for the layer in which R is installed is disabled when the CRAN mirror is changed.

```dockerfile
FROM ubuntu:focal

ENV R_VERSION=4.1.3
ENV TERM=xterm
ENV R_HOME=/usr/local/lib/R
ENV CRAN=https://packagemanager.rstudio.com/cran/__linux__/focal/latest
ENV TZ=Etc/UTC

COPY scripts/install_R.sh /rocker_scripts/install_R.sh

RUN /rocker_scripts/install_R.sh

ENV LANG=en_US.UTF-8

COPY scripts /rocker_scripts
```

In fact, however, the CRAN mirror settings are not used during the R installation; the CRAN mirror settings are only used when installing some R packages that are installed after the R installation.

Therefore, from the viewpoint of cache reuse, I suggest separating the part affected by the environment variable `CRAN` from `install_R.sh` to another script `setup_R.sh` and separating the layers as shown below.
This change allows the layer to be reused in cases where the base image is not changed and only the CRAN URL is changed (when setting the CRAN mirror date when a new version of R is released, or when changing the URL when binaries for `ubuntu:jammy` are made available in RSPM, as expected in R4.2.0).

```dockerfile
FROM ubuntu:focal

ENV R_VERSION=4.1.3
ENV TERM=xterm
ENV R_HOME=/usr/local/lib/R
ENV TZ=Etc/UTC

COPY scripts/install_R.sh /rocker_scripts/install_R.sh

RUN /rocker_scripts/install_R.sh

ENV LANG=en_US.UTF-8
ENV CRAN=https://packagemanager.rstudio.com/cran/__linux__/focal/latest

COPY scripts /rocker_scripts

RUN /rocker_scripts/setup_R.sh
```

To avoid confusion, it may be a good idea to change the name of the script that installs R from source to something like `install_R_source.sh`.

### LINKED ISSUE #390: fixing the BLAS hot-switching mechanism
There is a regression with the BLAS switching mechanism, as described here: https://www.rocker-project.org/use/other_topics/, 
in rocker-versioned2:

```shell
%docker run --rm rocker/verse:4.1.2 bash

R -q -e 'sessionInfo()' | grep so
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

update-alternatives --set libblas.so.3-x86_64-linux-gnu /usr/lib/x86_64-linux-gnu/blas/libblas.so.3

# nothing changed
R -q -e 'sessionInfo()' | grep so
BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so

# expected considering that
# ldd /usr/local/lib/R/lib/libR.so | grep blas
	libopenblas.so.0 => /lib/x86_64-linux-gnu/libopenblas.so.0 (0x00007f20400b0000)


```

This seems to be related to the auto-detection of the openblas lib during the R configuration, so that the `--with-blas="-lopenblas"` is used (cf https://cran.r-project.org/doc/manuals/r-devel/R-admin.html#OpenBLAS).

One simple way to fix that is to postpone the openblas installation AFTER the R compilation:

```diff
git diff scripts/install_R.sh
diff --git i/scripts/install_R.sh w/scripts/install_R.sh
index bad57c4..30f996b 100755
--- i/scripts/install_R.sh
+++ w/scripts/install_R.sh
@@ -24,11 +24,7 @@ export DEBIAN_FRONTEND=noninteractive
 R_HOME=${R_HOME:-/usr/local/lib/R}
 
 READLINE_VERSION=8
-OPENBLAS=libopenblas-dev
-if [ ${UBUNTU_VERSION} == "bionic" ]; then
-  READLINE_VERSION=7
-  OPENBLAS=libopenblas-dev
-fi
+
 
 apt-get update \
   && apt-get install -y --no-install-recommends \
@@ -46,7 +42,6 @@ apt-get update \
     libicu* \
     libpcre2* \
     libjpeg-turbo* \
-    ${OPENBLAS} \
     libpangocairo-* \
     libpng16* \
     libreadline${READLINE_VERSION} \
@@ -127,6 +122,14 @@ make
 make install
 make clean
 
+# Openblas installation: N.B: this must happen AFTER the R compilation otherwise the BLAS hot-swapping does not work
+OPENBLAS=libopenblas-dev
+if [ ${UBUNTU_VERSION} == "bionic" ]; then
+  READLINE_VERSION=7
+  OPENBLAS=libopenblas-dev
+fi
+apt-get install -y --no-install-recommends ${OPENBLAS}
+
 ## Add a default CRAN mirror
 echo "options(repos = c(CRAN = '${CRAN}'), download.file.method = 'libcurl')" >> ${R_HOME}/etc/Rprofile.site

```

then: 

```shell
docker build -f dockerfiles/r-ver_4.1.2.Dockerfile  -t r-ver_4.1.2_fixed  .
docker run --rm -ti r-ver_4.1.2_fixed bash

R -q -e 'sessionInfo()' | grep so
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/local/lib/R/lib/libRlapack.so

update-alternatives --set libblas.so.3-x86_64-linux-gnu /usr/lib/x86_64-linux-gnu/blas/libblas.so.3

R -q -e 'sessionInfo()' | grep so
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/local/lib/R/lib/libRlapack.so

# fixed because:
ldd /usr/local/lib/R/lib/libR.so | grep blas
	libblas.so.3 => /lib/x86_64-linux-gnu/libblas.so.3 (0x00007febf46aa000)
```

So AFAIU, this restores the previous and expected behaviour (for BLAS only, not for Lapack, as pointed out by Dirk).

### PR DESCRIPTION
**Pull request by eitsupi**
Close #385
Close #390

**ToDo**

- [x] Add `setup_R.sh` and move some lines from `install_R.sh` to `setup_R.sh`
- [x] Add a new GitHub Actions workflow to test building `rocker/r-ver`
- [x] Rename `install_R.sh` to `install_R_source.sh`
- [x] Merge `patch_install_command.sh` into `setup_R.sh`
- [x] Refactor `setup_R.sh`
- [x] Update Dockerfiles
- [x] Change BLAS installation order (#390)
- [x] Download R from the 0-Cloud CRAN mirror.
- [x] Update NEWS

### CODE REVIEW

**Comment by cboettig** [comment]
:+1: yup, using RStudio CDN seems like a better choice here.

**Comment by cboettig** [comment]
Also consider including proposed changes to blas from: https://github.com/rocker-org/rocker-versioned2/issues/390#issuecomment-1069180401

- [ ] `liblapack-dev` should be installed before R is compiled
- [ ] `libopenblas-dev` should install after R is compiled (I don't think this one is strictly necessary, but it makes sense).

File: scripts/install_R_source.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+make
+make install
+make clean
+
+## Install OpenBLAS after R is compiled
+## https://github.com/rocker-org/rocker-versioned2/issues/390
+ARCH=$(uname -m)
+apt-get install -y --no-install-recommends "${OPENBLAS}"
+update-alternatives --set "libblas.so.3-${ARCH}-linux-gnu" "/usr/lib/${ARCH}-linux-gnu/openblas-pthread/libblas.so.3"
```
Comment (eitsupi) [inline scripts/install_R_source.sh:138]: Wondering if these lines should be moved to `setup_R.sh`.

Comment (eitsupi) [reply scripts/install_R_source.sh]: I moved these lines to `setup_R.sh` in a1fbea402d66162b2eb6d2f96de65ed9a8238000.

**Comment by eitsupi** [comment]
I noticed that this modification increases the size by more than 30 MB...
I will add this to the news as well.

```shell
$ apt install --no-install-recommends liblapack-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  liblapack3
Suggested packages:
  liblapack-doc
The following NEW packages will be installed:
  liblapack-dev liblapack3
0 upgraded, 2 newly installed, 0 to remove and 47 not upgraded.
Need to get 6,395 kB of archives.
After this operation, 32.5 MB of additional disk space will be used.
Do you want to continue? [Y/n]
```

**Comment by eitsupi** [comment]
Now that the build test is working properly, I will merge this PR.

**Pull request merged**